File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.java
import java.util.*;

class Solution {
    
    ArrayList<Integer> num = new ArrayList<>();
    public int solution(int[] arr) {
        int count = 0;
    
        for(int i = 0; i<arr.length; i++){
            if(arr.length-i >= 3){
                plus(arr, i, arr[i]);
            }else{
                break;
            }
        }

        for (int a : num) {
            if (isPrime(a)) {
                count++;
            }
        }
        
        return count;
    }
    
    public void plus(int[] arr, int idx, int n){
        int number = n;
        
        for(int j = idx+1; j<arr.length; j++){
            if(arr.length-idx >= 3){
                
            }else{
                break;
            }
            
            for(int k = j+1; k<arr.length; k++){
                    number += arr[j]+arr[k];
                    num.add(number);
                    number = n;
            }
        }        
        
    }

    //소수인지 아닌지 판별해주기
    public boolean isPrime(int n) {
        if (n < 2) {
            return false;
        }

        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
let sum = [];

function plus(nums, idx, number){
    
    for(let i = idx+1; i < nums.length; i++){
        
        if(nums.length - i < 2){
            break;
        }
        
        for(let j = i+1; j<nums.length; j++){
            sum.push(nums[i]+nums[j]+number);
        }
    }
}

function isPrime(a){
    if(a <= 3){
        return true;
    }
    
    let count = 0;
    
    for(let i = 1; i <= a; i++){
        if(a % i == 0){
            count++;
        }
    }
    
    if(count == 2){
        return true;
    }else{
        return false;
    }
}

function solution(nums) {
    
    for(let i = 0; i<nums.length; i++){
        
        if(nums.length-i < 3){
            break;    
        }
        
        plus(nums, i, nums[i]);
    }
    
    let answer = 0;
    
    for(let a of sum){
        if(isPrime(a))
            answer++;
    }
    
    
    return answer;
}

